src/azure-cli/azure/cli/command_modules/eventhubs/custom.py

```diff
@@ -47,6 +70,67 @@ def cli_namespace_create(cmd, client, resource_group_name, namespace_name, locat
 
     return client.get(resource_group_name, namespace_name)
 
+class AlertAddEncryption(argparse._AppendAction):
+    def __call__(self, parser, namespace, values, option_string=None):
+        action = self.get_action(values, option_string)
+        super(AlertAddEncryption, self).__call__(parser, namespace, action, option_string)
+
+    def get_action(self, values, option_string):  # pylint: disable=no-self-use
+        config_length = len(values)
+        from azure.mgmt.eventhub.v2021_06_01_preview.models import KeyVaultProperties
+        from azure.mgmt.eventhub.v2021_06_01_preview.models import UserAssignedIdentityProperties
+
+        if 'key-name' not in [x.lower() for x in values]:
+            raise InvalidArgumentValueError('key-name and key-vault-uri are mandatory arguments for encryption configs.'.format(option_string))
+
+        if 'key-vault-uri' not in [x.lower() for x in values]:
+            raise InvalidArgumentValueError('key-name and key-vault-uri are mandatory arguments for encryption configs.'.format(option_string))
+
+        if config_length<4:
+            raise InvalidArgumentValueError('Please make sure that key-name and key-vault-uri arguments are present for every encryptionconfig'.format(option_string))
+
+        keyVaultObject = KeyVaultProperties()
+
+        if(values[0].lower()=='key-name'):
+            keyVaultObject.key_name = values[1]
+        else:
+            raise InvalidArgumentValueError('Arguements for encryption configs must be given in the order key-name key-vault-uri key-version. --encryption-config keyname [arg] keyvaulturi [arg] keyversion [arg]'.format(option_string))
+
+        if (values[2].lower() == 'key-vault-uri'):
+            keyVaultObject.key_vault_uri = values[3]
+            #remove trailing slash from keyvaulturi. KeyVault returns url with a trailing slash
+            if(keyVaultObject.key_vault_uri.endswith('/')):
+                keyVaultObject.key_vault_uri = keyVaultObject.key_vault_uri[:-1]
+        else:
+            raise InvalidArgumentValueError('Arguements for encryption configs must be given in the order key-name key-vault-uri key-version. --encryption-config keyname [arg] keyvaulturi [arg] keyversion [arg]'.format(option_string))
+
+        if (config_length>=6):
+            if (values[4].lower()=='key-version'):
```

Comment:
Instead of
```--encryption-config key-name key1 key-vault-uri https://mykeyvault.vault.azure.net/ key-version 0.0```,
you can design the param input as
```--encryption-config key-name=key1 key-vault-uri=https://mykeyvault.vault.azure.net/ key-version=0.0```

So that you can get the (key, value) pairs by splitting strings with `=`.
You don't need to guess if the fifth string is `key-version` or `user-assigned-identity`. And customers can pass in all info in any order

Reply:
Exactly, so it's required that no space around `=`. Till now, most of the params using this design works well.